riscv_analysis/src/cfg/ops.rs

```diff
@@ -84,22 +84,58 @@ impl MathOp {
                 let (x, y) = (x as u64, y as u64);
                 ((x * y) >> 32) as i32
             }
-            MathOp::Div => x / y,
-            MathOp::Divu => (x as u32 / y as u32) as i32,
-            MathOp::Rem => x % y,
-            MathOp::Remu => (x as u32 % y as u32) as i32,
+            // NOTE: The RISC-V spec doesn't trap for integer division by zero,
+            // instead, RISC-V returns the following results for x / 0 (or x % 0):
+            // - div   -1
+            // - divu: 2^32 - 1
+            // - rem:  x
+            // - remu: x
+            MathOp::Div => {
+                match y {
+                    0 => -1,    // 2^32 - 1 as i32
+                    _ => x / y,
+                }
+            },
```

Comment:
There are a set of methods called `checked_*` on i32 that can check for overflows and divide by zeros. It'll do the same thing, the syntax is shorter but could be harder to understand. I think the `match`, although longer, is easier to understand, so you pick which one you prefer.

```suggestion
            MathOp::Div => {
                x.checked_div(y).unwrap_or(-1)
            },
```